networkx/algorithms/coloring/greedy_coloring.py

```diff
@@ -65,14 +65,47 @@ def strategy_smallest_last(G, colors):
 
     """
     H = G.copy(with_data=False)
-
-    def min_and_pop():
-        """Removes and returns the minimum degree node in ``H``."""
-        node = min(H, key=H.degree)
-        H.remove_node(node)
-        return node
-
-    return reversed([min_and_pop() for x in G])
+    result = []
+
+    # Build initial degree list (i.e. the bucket queue data structure)
+    degrees = defaultdict(set)  # set(), for fast random-access removals
+    for node, d in H.degree():
+        degrees[d].add(node)
+        try:
+            H.lbound = min(H.lbound, d)  # Lower bound on min-degree.
+        except AttributeError:
+            # Python 2.X doesn't support `nonlocal`.
+            # Workaround: set as attribute of the smallest-last graph.
+            H.lbound = d
+
+    def find_min_degree():
```

Comment:
When you eliminate `lbound`, it obfuscates two sections of code.

```python
for d in itertools.count(min_degree):
    if d in degrees:
        min_degree = d
        break
```
In this block, we're counting up from `min_degree` until we find the minimum degree, so it's not the best label for this variable. 

Further, in `min_degree -= 1`, it's not strictly true that the minimum degree of the graph decreases. It truly is the lower bound that decreases, and using `min_degree` as double-duty is less clear to readers.